api/specs/webserver/scripts/openapi_projects_nodes.py

```diff
@@ -53,6 +56,34 @@ async def get_project_services_access_for_gid(project_id: ProjectID, for_gid: Gr
 )
 
 
-if __name__ == "__main__":
+@app.get(
+    "/projects/{project_id}/nodes/-/homepage",
+    response_model=Envelope[list[_ProjectNodeHomePage]],
+    tags=TAGS,
+    operation_id="list_project_node_homepages",
+    summary="Lists all nodes homepages info in the project",
+)
+async def list_project_node_homepages(project_id: ProjectID):
+    ...
 
+
+assert_handler_signature_against_model(list_project_node_homepages, ProjectPathParams)
+
+
+@app.get(
+    "/projects/{project_id}/nodes/{node_id}/homepage",
```

Comment:
what is a node homepage? in case of service with a webserver... is that the root entry of that service?

Reply:
Following @matusdrobuliak66  suggestion I changed that to `preview`. 

A preview is an object associated to the underlying node that for the moment contains screenshots but can be extended to contain other information associated to the underlying service. The content of this object is generated by the service vendor or user.